nabs/plans.py

```diff
@@ -653,3 +659,138 @@ def daq_a3scan(m1, a1, b1, m2, a2, b2, m3, a3, b3, nsteps):
     """
 
     yield from bp.scan([], m1, a1, b1, m2, a2, b2, m3, a3, b3, nsteps)
+
+
+def fixed_target_scan(sample, detectors, x_motor, y_motor, scan_motor, ss,
+                      n_shots, path, snake_like=True):
+    """
+    Scan over two variables in steps simultaneously.
+
+    This is a `nabs` version of ``bluesky``'s built-in
+    `bluesky.plans.list_scan` plan.
+    This scan is designed to be used with 3 motors, the x and y motors that
+    move to a designaget target on a sample wafer, and one scan_motor that can
+    be delay time, laser power or some other motor positions.
+
+    Parameters
+    ----------
+    sample : str
+        The name of the sample we're interested in.
+    detectors : list
+        Objects to read into Python in the scan.
+    x_motor : obj
+        Motor object corresponding to the x axes.
+    y_motor : obj
+        Motor object corresponding to the y axes.
+    scan_motor : obj
+        The motor being scanned. It can be e.g., delay time, laser power, some
+        other motor position, etc.
+    ss : list
+        List of all the points (samples) for the scan_motor to go through.
+    n_shots : int
+        Indicates how many shots should be taken, or how many samples should
+        be scanned on the grid.
+    path : str
+        Path where the sample file is located.
+    snake_like : bool, optional
+        Indicates if the scanning should be following a snake-like pattern.
+        Defaluts to `True`.
+    """
+
+    # TODO: remember what targets have been shot -
+    # display this in a little window - bit mapping?
+
+    global _last_index
+    detectors = list(detectors) + [scan_motor]
+
+    rows, columns, last_shot_index, xx, yy = get_sample_info(sample, path)
+    if snake_like:
+        xx = snake_grid_list(np.array(xx).reshape(rows, columns))
+
+    _last_index = last_shot_index
+    if (last_shot_index + (n_shots * len(ss))) >= len(xx):
+        raise IndexError('The number of last_shot_index + n_shots * len(ss): '
+                         f'{last_shot_index} + {n_shots} * {len(ss)}= '
+                         f'{last_shot_index + (n_shots * len(ss))}: '
+                         f' bigger than the available samples: {len(xx)}. '
+                         'Please provide a number in range.')
+
+    def inner_scan():
+        global _last_index
+        try:
+            yield from bps.open_run(md={})
+            for i in range(len(ss)):
+                yield from bps.mv(scan_motor, ss[i])
+                yield from bps.read(scan_motor)
```

Comment:
I think this line no longer does anything

Reply:
Yes that is correct - thank you, I removed it in 7ac3131